### PR TITLE
Separate network elements info types (the second part)

### PR DESCRIPTION
**Pull request by AAJELLAL**

### CODE REVIEW

File: src/main/java/org/gridsuite/network/map/dto/vscconverterstation/VscConverterStationTabInfos.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ */
+package org.gridsuite.network.map.dto.vscconverterstation;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.iidm.network.VscConverterStation;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+/**
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/network/map/dto/vscconverterstation/VscConverterStationTabInfos.java:15]: Esthetical : could you lift the empty line above your the javadoc block, here and some other files ?

Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/vscconverterstation/VscConverterStationTabInfos.java:15]: Even if it's checkstyle validated?

Comment (laurgarn) [reply src/main/java/org/gridsuite/network/map/dto/vscconverterstation/VscConverterStationTabInfos.java:15]: Checkstyle has limits too ;-)

Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/vscconverterstation/VscConverterStationTabInfos.java:16]: Done

File: src/main/java/org/gridsuite/network/map/NetworkMapController.java [hunk 2]
```diff
@@ -146,121 +56,26 @@ public class NetworkMapController {
...
-    @GetMapping(value = "/networks/{networkUuid}/voltage-levels/{voltageLevelId}", produces = APPLICATION_JSON_VALUE)
-    @Operation(summary = "Get voltage level description")
-    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Voltage level description")})
-    public @ResponseBody VoltageLevelMapData getVoltageLevel(@Parameter(description = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
-                                                                     @Parameter(description = "Voltage level id") @PathVariable("voltageLevelId") String voltageLevelId,
-                                                                     @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId) {
-        return networkMapService.getVoltageLevel(networkUuid, variantId, voltageLevelId);
+    @GetMapping(value = "/networks/{networkUuid}/elements/{elementId}", produces = APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get a network element")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Elements description")})
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/NetworkMapController.java:72]: Typo: Element description

File: src/main/java/org/gridsuite/network/map/dto/ElementInfos.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    private String name;
+
+    public static ElementInfos toData(Identifiable<?> identifiable) {
+        ElementInfos.ElementInfosBuilder builder = builder()
+                .name(identifiable.getOptionalName().orElse(null))
+                .id(identifiable.getId());
+        return builder.build();
+    }
+
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/dto/ElementInfos.java:47]: I think all these following utility methods should be put in a separate utility class, and not in this dto

File: src/main/java/org/gridsuite/network/map/dto/ElementInfos.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    protected static CurrentLimitsData toMapDataCurrentLimits(CurrentLimits limits) {
+        CurrentLimitsData.CurrentLimitsDataBuilder builder = CurrentLimitsData.builder();
+        boolean empty = true;
+        if (!Double.isNaN(limits.getPermanentLimit())) {
+            builder.permanentLimit(limits.getPermanentLimit());
+            empty = false;
+        }
+        if (limits.getTemporaryLimits() != null && !limits.getTemporaryLimits().isEmpty()) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/dto/ElementInfos.java:59]: Use CollectionUtils.isEmpty

File: src/main/java/org/gridsuite/network/map/dto/voltagelevel/VoltageLevelMapInfos.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    private String substationId;
+
+    private double nominalVoltage;
+
+    public static VoltageLevelMapInfos toData(Identifiable<?> identifiable) {
+        VoltageLevel voltageLevel = (VoltageLevel) identifiable;
+        return VoltageLevelMapInfos.builder()
+                .id(voltageLevel.getId())
+                .name(voltageLevel.getOptionalName().orElse(null))
+                .substationId(voltageLevel.getSubstation().orElseThrow().getId())
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/dto/voltagelevel/VoltageLevelMapInfos.java:32]: if voltageLevel has no substation, we should no throw an exception ...

File: src/main/java/org/gridsuite/network/map/dto/BusbarSectionFinderTraverser.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/**
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/dto/BusbarSectionFinderTraverser.java:1]: This class is not a DTO : it should be in another package

File: src/main/java/org/gridsuite/network/map/dto/battery/BatteryTabInfos.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        Battery battery = (Battery) identifiable;
+        Terminal terminal = battery.getTerminal();
+        BatteryTabInfos.BatteryTabInfosBuilder builder = BatteryTabInfos.builder()
+                .name(battery.getOptionalName().orElse(null))
+                .id(battery.getId())
+                .terminalConnected(terminal.isConnected())
+                .voltageLevelId(terminal.getVoltageLevel().getId())
+                .targetP(battery.getTargetP())
+                .targetQ(battery.getTargetQ());
+        if (!Double.isNaN(terminal.getP())) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/dto/battery/BatteryTabInfos.java:51]: you can more simply use :    .p(nullIfNan(terminal.getP())
                                             .q(nullIfNan(terminal.getQ())

same remarks in other dtos ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/network/map/dto/ElementInfos.java (hunk 3 above)
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/ElementInfos.java:47]: Same remark as in my previous response.

File: src/main/java/org/gridsuite/network/map/dto/voltagelevel/VoltageLevelMapInfos.java (hunk 5 above)
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/voltagelevel/VoltageLevelMapInfos.java:32]: Same remark as in my previous response

File: src/main/java/org/gridsuite/network/map/dto/BusbarSectionFinderTraverser.java (hunk 6 above)
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/BusbarSectionFinderTraverser.java:1]: Same remark as in my previous response

File: src/main/java/org/gridsuite/network/map/dto/battery/BatteryTabInfos.java (hunk 7 above)
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/battery/BatteryTabInfos.java:51]: I just copy/paste the original code.But indeed it could be improved.

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 8]
```diff
@@ -1085,330 +156,59 @@ private List<String> getEquipmentsIds(UUID networkUuid, String variantId, List<S
...
-                network.getSubstation(id).getVoltageLevelStream().forEach(v ->
-                    v.getConnectables(ThreeWindingsTransformer.class).forEach(t -> res.add(toMapData(t)))));
-            return res.stream().collect(Collectors.toList());
-        }
-    }
-
     public AllMapData getAll(UUID networkUuid, String variantId, List<String> substationsId) {
         Network network = getNetwork(networkUuid, getPreloadingStrategy(substationsId), variantId);
 
         if (substationsId == null) {
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/NetworkMapService.java:162]: Done

File: src/main/java/org/gridsuite/network/map/dto/threewindingstransformer/ThreeWindingsTransformerTabInfos.java [hunk 9]
```diff
@@ -1,46 +1,43 @@
...
     @JsonInclude(JsonInclude.Include.NON_NULL)
-    private VoltageLevelMapData voltageLevel1;
+    private VoltageLevelListInfos voltageLevel1;
 
     //TODO put this into the DTO voltageLevel1
     private String voltageLevelId1;
 
     @JsonInclude(JsonInclude.Include.NON_NULL)
-    private VoltageLevelMapData voltageLevel2;
+    private VoltageLevelListInfos voltageLevel2;
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/threewindingstransformer/ThreeWindingsTransformerTabInfos.java:31]: Done

File: src/main/java/org/gridsuite/network/map/dto/threewindingstransformer/ThreeWindingsTransformerTabInfos.java [hunk 10]
```diff
@@ -1,46 +1,43 @@
...
 
     //TODO put this into the DTO voltageLevel2
     private String voltageLevelId2;
 
     @JsonInclude(JsonInclude.Include.NON_NULL)
-    private VoltageLevelMapData voltageLevel3;
+    private VoltageLevelListInfos voltageLevel3;
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/threewindingstransformer/ThreeWindingsTransformerTabInfos.java:37]: Done

File: src/main/java/org/gridsuite/network/map/dto/twowindingstransformer/TwoWindingsTransformerTabInfos.java [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+
+/**
+ * @author AJELLAL Ali <ali.ajellal@rte-france.com>
+ */
+@SuperBuilder
+@Getter
+public class TwoWindingsTransformerTabInfos extends AbstractTwoWindingsTransformerInfos {
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    private VoltageLevelListInfos voltageLevel1;
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/twowindingstransformer/TwoWindingsTransformerTabInfos.java:28]: Done

File: src/main/java/org/gridsuite/network/map/dto/twowindingstransformer/TwoWindingsTransformerTabInfos.java [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+
+    //TODO put this into the DTO voltageLevel1
+    private String voltageLevelId1;
+
+    //TODO put this into the DTO voltageLevel1
+    private String voltageLevelName1;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    private VoltageLevelListInfos voltageLevel2;
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/dto/twowindingstransformer/TwoWindingsTransformerTabInfos.java:37]: Done

File: src/main/java/org/gridsuite/network/map/model/AllMapData.java [hunk 13]
```diff
@@ -17,17 +18,18 @@
 @Builder
 @Getter
 public class AllMapData {
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/model/AllMapData.java:20]: Done

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 14]
```diff
@@ -1085,330 +156,59 @@ private List<String> getEquipmentsIds(UUID networkUuid, String variantId, List<S
...
-        } else {
-            Set<ThreeWindingsTransformerMapData> res = new LinkedHashSet<>();
-            substationsId.stream().forEach(id ->
-                network.getSubstation(id).getVoltageLevelStream().forEach(v ->
-                    v.getConnectables(ThreeWindingsTransformer.class).forEach(t -> res.add(toMapData(t)))));
-            return res.stream().collect(Collectors.toList());
-        }
-    }
-
     public AllMapData getAll(UUID networkUuid, String variantId, List<String> substationsId) {
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/main/java/org/gridsuite/network/map/NetworkMapService.java:159]: Done

File: src/main/java/org/gridsuite/network/map/NetworkMapController.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/NetworkMapController.java:72]: Done

File: src/main/java/org/gridsuite/network/map/dto/ElementInfos.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/dto/ElementInfos.java:47]: Done

File: src/main/java/org/gridsuite/network/map/dto/ElementInfos.java (hunk 4 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/dto/ElementInfos.java:59]: Done

File: src/main/java/org/gridsuite/network/map/dto/BusbarSectionFinderTraverser.java (hunk 6 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/dto/BusbarSectionFinderTraverser.java:1]: Done

File: src/main/java/org/gridsuite/network/map/dto/battery/BatteryTabInfos.java (hunk 7 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/dto/battery/BatteryTabInfos.java:51]: Done

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 15]
```diff
@@ -1568,4 +383,15 @@ public List<String> getEquipmentsIds(UUID networkUuid, String variantId, List<St
...
         }
     }
+
+    private Identifiable getIdentifiable(Network network, ElementType elementType, String elementId) {
+        switch (elementType) {
+            case TWO_WINDINGS_TRANSFORMER:
+                return network.getTwoWindingsTransformer(elementId);
+            case GENERATOR:
+                return network.getGenerator(elementId);
+            default:
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/NetworkMapService.java:393]: Done

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/NetworkMapService.java:162]: Done

**Pull request closed without merging**